main.rb

```diff
@@ -0,0 +1,220 @@
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/ModuleLength, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |k|
+      yield to_a[k]
+    end
+    to_a
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |k|
+      yield to_a[k], k
+    end
+    to_a
+  end
+
+  def my_select
+    return enum_for(:my_each) unless block_given?
+
+    result = []
+    0.upto(to_a.size - 1) do |k|
+      ret = yield to_a[k]
+      result << to_a[k] if ret == true
+    end
+    result
+  end
+
+  def my_all?(*args)
+    if block_given?
+      to_a.my_each do |item|
+        return false unless yield item
+      end
+    elsif args&.length&.positive? and to_a.length.positive?
+      if args[0].class == Regexp
+        to_a.my_each do |item|
+          return false unless item.to_s =~ args[0]
+        end
+      elsif args[0].class == Class
+        to_a.my_each do |item|
+          return false unless item.is_a?args[0]
+        end
+      else
+        to_a.my_each do |item|
+          return false unless item == args[0]
+        end
+      end
+    else
+      to_a.my_each do |item|
+        return false unless item
+      end
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if block_given?
+      to_a.my_each do |item|
+        return true if yield item
+      end
+    elsif args&.length&.positive? and to_a.length.positive?
+      if args[0].class == Regexp
+        to_a.my_each do |item|
+          return true if item.to_s =~ args[0]
+        end
+      elsif args[0].class == Class
+        to_a.my_each do |item|
+          return true if item.is_a?args[0]
+        end
+      else
+        to_a.my_each do |item|
+          return true if item == args[0]
+        end
+      end
+    else
+      to_a.my_each do |item|
+        return true if item
+      end
+    end
+    false
+  end
+
+  def my_none?(*args)
+    if block_given?
+      to_a.my_each do |item|
+        return true unless yield item
+      end
+    elsif args&.length&.positive? and to_a.length.positive?
+      if args[0].class == Regexp
+        to_a.my_each do |item|
+          return true unless item.to_s =~ args[0]
+        end
+      elsif args[0].class == Class
+        to_a.my_each do |item|
+          return true unless item.is_a?args[0]
+        end
+      else
+        to_a.my_each do |item|
+          return true unless item == args[0]
+        end
+      end
+    else
+      to_a.my_each do |item|
+        return true unless item
+      end
+    end
+    false
+  end
+
+  def my_count(*args)
```

Comment:
- compare results in both cases

```ary = [1, 2, 4, 2]
   ary.my_count(2, &:even?) != ary.count(2, &:even?)
```
I think the right behavior here ought to be tested by ```ary.count(&:even?)```
 in our case ```ary.my_count(&:even?)``` I am raising this issue for you to go back to the functionalities of the method so you can be clear on it.